src/main/java/com/netgrif/application/engine/rules/service/RuleEngine.java

```diff
@@ -1,66 +1,74 @@
 import com.netgrif.application.engine.rules.service.interfaces.IRuleEngine;
 import com.netgrif.application.engine.workflow.domain.Case;
 import com.netgrif.application.engine.workflow.domain.Task;
+import lombok.extern.slf4j.Slf4j;
 import org.kie.api.runtime.KieSession;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Lookup;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
 import java.util.Arrays;
 import java.util.List;
 
+@Slf4j
 @Service
 public abstract class RuleEngine implements IRuleEngine {
 
-    private static final Logger log = LoggerFactory.getLogger(RuleEngine.class);
-
     @Value("${rule-engine.rethrow-exceptions:#{false}}")
     protected boolean rethrowExceptions;
 
+    @Autowired
+    private RuleRepository ruleRepository;
+
     @Lookup
     protected abstract KieSession ruleEngine();
 
     @Override
-    public void evaluateRules(Case useCase, CaseCreatedFact caseCreatedFact) {
-        evaluateWithFacts(Arrays.asList(useCase, caseCreatedFact));
+    public int evaluateRules(Case useCase, CaseCreatedFact caseCreatedFact) {
+        return evaluateWithFacts(Arrays.asList(useCase, caseCreatedFact));
     }
 
     @Override
-    public void evaluateRules(Case useCase, Task task, TransitionEventFact transitionEventFact) {
-        evaluateWithFacts(Arrays.asList(useCase, task, transitionEventFact));
+    public int evaluateRules(Case useCase, Task task, TransitionEventFact transitionEventFact) {
+        return evaluateWithFacts(Arrays.asList(useCase, task, transitionEventFact));
     }
 
     @Override
-    public void evaluateRules(Case useCase, ScheduledRuleFact scheduledRuleFact) {
-        evaluateWithFacts(Arrays.asList(useCase, scheduledRuleFact));
+    public int evaluateRules(Case useCase, ScheduledRuleFact scheduledRuleFact) {
+        return evaluateWithFacts(Arrays.asList(useCase, scheduledRuleFact));
     }
 
     @Override
-    public void evaluateRules(PetriNet petriNet, NetImportedFact fact) {
-        evaluateWithFacts(Arrays.asList(petriNet, fact));
+    public int evaluateRules(PetriNet petriNet, NetImportedFact fact) {
+        return evaluateWithFacts(Arrays.asList(petriNet, fact));
     }
 
     @Override
-    public void evaluateRules(PetriNet petriNet, ScheduledRuleFact scheduledRuleFact) {
-        evaluateWithFacts(Arrays.asList(petriNet, scheduledRuleFact));
+    public int evaluateRules(PetriNet petriNet, ScheduledRuleFact scheduledRuleFact) {
+        return evaluateWithFacts(Arrays.asList(petriNet, scheduledRuleFact));
     }
 
-
-    private void evaluateWithFacts(List<Object> facts) {
+    private int evaluateWithFacts(List<Object> facts) {
+        if (ruleRepository.count() == 0) {
+            return 0;
+        }
         KieSession session = null;
+        int numberOfRulesExecuted = 0;
```

Comment:
Primitive type `int` has default value `0`, so it is not necessary to intialize.

Reply:
It is just for readability